src/main/scala/org/broadinstitute/gpp/poolq3/barcode/BarcodePolicy.scala

```diff
@@ -243,3 +243,56 @@ object TemplatePolicy {
   }
 
 }
+
+final case class SplitBarcodePolicy(
+  prefix1: String,
+  b1Length: Int,
+  gap: Int,
+  prefix2: String,
+  b2Length: Int,
+  minPrefix1StartPos: Option[Int],
+  maxPrefix1StartPos: Option[Int] = None
+) extends BarcodePolicy {
+
+  private[this] val minPrefix1StartPosInt: Int = minPrefix1StartPos.getOrElse(0)
+  private[this] val maxPrefix1StartPosInt: Int = maxPrefix1StartPos.getOrElse(Int.MaxValue)
+  private[this] val p1Length = prefix1.length
+  private[this] val p2Length = prefix2.length
+  private[this] val expectedP2Offset = prefix1.length + b1Length + gap
+  private[this] val patternLength = expectedP2Offset + p2Length + b2Length
+
+  override def length: Int = b1Length + b2Length
+
+  override def find(read: Read): Option[FoundBarcode] = {
+    val p1Index = read.seq.indexOf(prefix1, minPrefix1StartPosInt)
+    if (p1Index < 0) None
+    else if (p1Index > maxPrefix1StartPosInt) None
+    else {
+      // now we need to check if the read is long enough to contain the whole pattern starting at index
+      val patternEnd = p1Index + patternLength
+      if (read.seq.length < patternEnd) None
+      else {
+        val p2Index = p1Index + expectedP2Offset
+        val rsca = read.seq.toCharArray()
```

Comment:
I think you can avoid this whole array copy if you use `String#charAt()` for your array access in `containsP2()` and then use `String.getChars()` in place of the subsequent array copies to populate `dest`.  Hard to know how all that would affect performance but it would be interesting to benchmark the difference.

Reply:
Yes, I removed the `toCharArray()` in a later commit but I actually didn't know about `String#charAt` and I'm going to add a commit using that. More on benchmarking later.